checkstyle-tester/diff.groovy

```diff
@@ -53,25 +61,36 @@ def getCliOptions(args) {
     def cliOptionsDescLineLength = 120
     def cli = new CliBuilder(usage:'groovy diff.groovy [options]', header: 'options:', width: cliOptionsDescLineLength)
     cli.with {
-        r(longOpt: 'localGitRepo', args: 1, required: true, argName: 'localGitRepo', 'Path to local git repository (required)')
-        b(longOpt: 'baseBranch', args: 1, required: false, argName: 'baseBranch', 'Base branch name. Default is master (optional, default is master)')
-        p(longOpt: 'patchBranch', args: 1, required: true, argName: 'patchBranch', 'Name of the patch branch in local git repository (required)')
-        c(longOpt: 'checkstyleCfg', args: 1, required: true, argName: 'checkstyleCfg', 'Path to checkstyle config file (required)')
-        l(longOpt: 'listOfProjects', args: 1, required: true, argName: 'listOfProjects', 'Path to file which contains projects to test on (required)')
+        r(longOpt: 'localGitRepo', args: 1, required: true, argName: 'path', 'Path to local git repository (required)')
+        b(longOpt: 'baseBranch', args: 1, required: false, argName: 'branch_name', 'Base branch name. Default is master (optional, default is master)')
+        p(longOpt: 'patchBranch', args: 1, required: true, argName: 'branch_name', 'Name of the patch branch in local git repository (required)')
+        bc(longOpt: 'baseConfig', args: 1, required: false, argName: 'path', 'Path to the base checkstyle config file (required if patchConfig is specified)')
+        pc(longOpt: 'patchConfig', args: 1, required: false, argName: 'path', 'Path to the patch checkstyle config file (required if baseConfig is specified)')
+        c(longOpt: 'config', args: 1, required: false, argName: 'path', 'Path to the checkstyle config file (required if baseConfig and patchConfig are not secified)')
+        l(longOpt: 'listOfProjects', args: 1, required: true, argName: 'path', 'Path to file which contains projects to test on (required)')
     }
     return cli.parse(args)
 }
 
 def areValidCliOptions(cliOptions) {
     def valid = true
-    def localGitRepo = new File(cliOptions.localGitRepo)
-    def patchBranch = cliOptions.patchBranch
-    def baseBranch = cliOptions.baseBranch
-    if (!isValidGitRepo(localGitRepo)
-           || !isExistingGitBranch(localGitRepo, patchBranch)
-           || baseBranch && !isExistingGitBranch(localGitRepo, baseBranch)) {
+    def baseConfig = cliOptions.baseConfig
+    def patchConfig = cliOptions.patchConfig
+    def config = cliOptions.config
+    if ((baseConfig || patchConfig || !config)
+            && (!baseConfig || !patchConfig || config)) {
         valid = false
```

Comment:
Shouldn't we print what error is like we do in other `isValid`/`isExisting` methods below?